Fix part of #3602: Add accessibility label for ProfileEditActivity

kkmurerwa opened the pull request:
<!-- READ ME FIRST: Please fill in the explanation section below and check off every point from the Essential Checklist! -->
## Explanation
<!--
  - Explain what your PR does. If this PR fixes an existing bug, please include
  - "Fixes #bugnum:" in the explanation so that GitHub can auto-close the issue
  - when this PR is merged.
  -->
Fix part of #3602: When merged, this PR will;
- Create a strings item in vaues/strings.xml with the accessibility text label for the ProfileEditActivity.
- Add a label for the activity in the manifest file.
- Remove the activity from the accessibility_label_exemptions.textproto file.
- Add test to check that the appropriate label is displayed for ProfileEditActivity.
- Modified tests in ProfileEditActivityTest to reflect new title behavior and remove any checks for the username as the toolbar title.
- Removed deprecated testProfileEdit_updateName_checkNewNameDisplayed() test that checked for previous title behaviour.

New changes after 22/03/23 review;
- Rename the string from R.string.profile_edit_activity_label to R.string.profile_edit_activity_title
- Update the Manifest file to reflect this change.
- Modify the ProfileEditActivityTest class to reflect the change and test for the toolbar title.
- Linting check fixes.

## Essential Checklist
<!-- Please tick the relevant boxes by putting an "x" in them. -->
- [x] The PR title and explanation each start with "Fix #bugnum: " (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] Any changes to [scripts/assets](https://github.com/oppia/oppia-android/tree/develop/scripts/assets) files have their rationale included in the PR explanation.
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary code changes from Android Studio ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes)).
- [x] The PR is made from a branch that's **not** called "develop" and is up-to-date with "develop".
- [x] The PR is **assigned** to the appropriate reviewers ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).

kkmurerwa committed:
Fix: Add accessibility label for ProfileEditActivity

When merged, this commit will;
- Create a strings item in R.strings with the accessibility text for the activity.
- Add a label for the activity in the manifest file.
- Remove the activity from the accessibility_label_exemptions.textproto file.

adhiamboperes commented:
Thanks @kkmurerwa!
I have a two comments:
- The PR explanation should start with "Fix part of #3602" so you can link this PR to the issue. (Also, you added the explanation above the heading instead of below it)
- Could you add a `testProfileEditActivity_hasCorrectActivityLabel()` test for this change?

adhiamboperes reviewed (changes requested):
Thanks @kkmurerwa!
I have a two comments:

- The PR explanation should start with "Fix part of https://github.com/oppia/oppia-android/issues/3602" so you can link this PR to the issue. (Also, you added the explanation above the heading instead of below it)
- Could you edit the testProfileEditActivity_hasCorrectActivityLabel() test for this change?

kkmurerwa committed:
test: Add test to check that the appropriate label is displayed for ProfileEditActivity

kkmurerwa committed:
chore: Remove unused import on ProfileEditActivityTest.kt

kkmurerwa committed:
chore: Reorganize imports to be in lexicographic order

kkmurerwa committed:
fix: Fix tests in ProfileEditActivityTest

kkmurerwa committed:
chore: Clean up the imports to be in line with linting guidelines

kkmurerwa committed:
chore: Delete testProfileEdit_updateName_checkNewNameDisplayed() test that checked for previous title behaviour

kkmurerwa committed:
merge: Merge from develop

adhiamboperes reviewed (approved):
LGTM

BenHenning committed:
Merge branch 'develop' into add-label-for-profileeditactivity

BenHenning commented on app/src/sharedTest/java/org/oppia/android/app/settings/profile/ProfileEditActivityTest.kt:
```diff
@@ -346,24 +338,6 @@ class ProfileEditActivityTest {
   }
 
-  @Test
-  fun testProfileEdit_updateName_checkNewNameDisplayed() {
```
I'm a bit confused by this change--isn't the PR only changing the activity's accessibility label not its title?

kkmurerwa replied:
It is changing both the title and the label. Sean suggested we change the behaviour of the activity so that the title shown is **_Edit Profile_** and not the name of the profile being displayed.

kkmurerwa replied:
Here is a brief snapshot of the conversation I had with Sean about the issue.

> Kenneth Murerwa, Thu 5:24 PM
> Hey @Sean Lip and @Ben Henning. I am having a bit of an issue here. I am currently working on issue #3602 (https://github.com/oppia/oppia-android/issues/3602) and while writing a test for the issue, I realized a conflict in the activity behavior.  Currently, the ProfileEditActivity has the name of the profile being edited on the action bar instead of the label/title (screenshot attached). If I write a test to check that the label is displayed, it always fails since the title is overwritten by the profile name. Is this the accepted behavior (and therefore I should not make any changes to the test since a test for that already exists) or should I ensure that the label (currently Edit Profile) should be shown instead of the profile name?
> 
> ![Screenshot of the current screen](https://user-images.githubusercontent.com/18438114/224654121-2fd9d818-1ac6-4190-bf34-ae2ca2652959.png)
> 
> Sean Lip, Thu 5:31 PM
> Let's show Edit Profile in the navbar, since the profile name is already shown just below it. Thanks for checking!

BenHenning replied:
I think it's fine that we make the title & label the same, but we should then be following the approach similar to here: https://github.com/oppia/oppia-android/blob/087f2fe85aa01655ad03bbebbc64ab5edd13089f/app/src/sharedTest/java/org/oppia/android/app/policies/PoliciesActivityTest.kt#L173 where:
- The string name is "title" not "label"
- The test verifies the toolbar title

BenHenning replied:
This looks to be resolved in the latest changes.

BenHenning reviewed (changes requested):
Thanks @kkmurerwa! Just had one comment looking for clarity. Are we actually changing the title in this PR or just the accessibility label? Bit confused about the test changes here.

adhiamboperes commented:
Hi @kkmurerwa, when you'd like someone to take the next action on your PR you can mention them by @xxx PTAL and then oppiabot should assign them for you. Also review the wiki [here](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section) on general code review guidelines.

kkmurerwa commented:
@BenHenning PTAL

BenHenning reviewed (commented):
Thanks @kkmurerwa. Had one follow-up--PTAL.

kkmurerwa committed:
Merge branch 'develop' into add-label-for-profileeditactivity

kkmurerwa committed:
fix: Update the profile_edit_activity_label string
- Rename the string from R.string.profile_edit_activity_label to R.string.profile_edit_activity_title
- Update the Manifest file to reflect this change.
- Modify the ProfileEditActivityTest class to reflect the change and test for toolbar title.

kkmurerwa committed:
fix: Fix lint issues with ProfileEditActivityTest

kkmurerwa committed:
fix: Try fix for the lint checks error

kkmurerwa committed:
fix: Try fix for the lint checks error

kkmurerwa committed:
fix: Try fix for the lint checks error by reorganizing the test method

kkmurerwa commented:
Hey @BenHenning. I renamed the string, modified the manifest and the relevant test. I hope everything is good now. I have updated the PR comment with the changes made under today's date (22/03/23). PTAL and let me know if there are any further changes needed.

kkmurerwa commented:
@BenHenning PTAL

BenHenning reviewed (changes requested):
Thanks @kkmurerwa! This LGTM!

One note: please make sure to reply directly to all comments between review passes (to ensure it's explicitly clear that you've addressed the comments).

BenHenning reviewed (approved):
Oops, meant to approve.
